src/makeInfinityAdder.js

```diff
@@ -3,8 +3,24 @@
 /***
  * @return {function}
  */
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(a = 0) {
+  let sum = a;
+
+  function add(b) {
+    if (b === undefined) {
+      const result = sum;
+
+      sum = 0;
```

Comment:
Resetting `sum` to `0` after returning the result is correct if the function is intended to start a new sum after each complete addition cycle. If the intention is to continue accumulating, consider removing this reset.